### PR TITLE
[design-system] Typography v2 styles and components

### LINKED ISSUE #99: [design system] Add new core typography styles
**What needs to be done? Why does it need to be done?**
Type styles as seen in [the v2 designs](https://www.figma.com/file/v7ei7OnUkPkwJO6ukOvTjl/System-v2?node-id=6%3A6) should be defined and exported as SCSS mixins, JS string constants, and Styled Components.

**Additional context**
SCSS should be adapted from `pulse-dashboard`, which already implements most of what is needed (but with a different sans typeface)

### LINKED ISSUE #101: Integrate type styles with all components
**What needs to be done? Why does it need to be done?**
All components that are expected to reflect certain type styles should explicitly those styles or components as defined in #99.

**Additional context**
Some existing components apply their own type styles, some refer to the theme props, and some simply depend on external or global style rules. All of these strategies should be removed in favor of using the new styles and components directly.

### LINKED ISSUE #102: Remove theme requirement
**What needs to be done? Why does it need to be done?**
Based on the latest iteration of the design system, we no longer need to support theming at this time. Any remaining references to the theme should be removed.

**Additional context**
n/a

### LINKED ISSUE #103: New major version release
**What needs to be done? Why does it need to be done?**
As the new typography system will be a breaking and potentially disruptive change, it necessitates a major version bump. This should be thoroughly tested with the pulse-dashboard and Spotlight applications before its publication, with revisions to this package as necessary, before it is published. Upgrades to dependent applications should preferably be prepared and ready for merge ASAP after package publication to minimize the potential for application features to be blocked on this major version bump.

**Additional context**
n/a

### PR DESCRIPTION
**Pull request by macfarlandian**
## Description of the change

This is the complete set of changes required to implement the new typography styles and publish a new major version; due to the dependencies between projects it didn't make much sense to try to separate this into multiple pull requests. It is big, though, so the commits are reviewable individually and each corresponds to a ticket that describes a logical chunk of work. The meatiest one is 6689441b0114831a2104df9772fa932859a01d99 but most of the SCSS is refactored and lightly updated from `pulse-dashboard` and most of the `styled-components` code is glorified boilerplate to make use of the SCSS.

Storybook (latest alpha, which is identical but for package version): https://recidiviz.github.io/web-libraries/?path=/story/design-system-pages-typography--ui and related pages as seen in sidebar

The real test of this is in the applications themselves, which I invite you to review and QA rigorously: each has been published to a preview channel (linked from their respective PRs).

Spotlight integration: https://github.com/Recidiviz/public-dashboard/pull/579
Pulse Dashboard integration: https://github.com/Recidiviz/pulse-dashboard/pull/1986

## Type of change

- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

closes #99
closes #101 
closes #102 
closes #103 

### Development

These boxes should be checked by the submitter prior to merging:

- [x] Manual testing against realistic data has been performed locally

### Code review

These boxes should be checked by reviewers prior to merging:

- [ ] This pull request has a descriptive title and information useful to a reviewer
- [ ] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [ ] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

**Review by jovergaag — APPROVED** [review]
Amazing, I love it.

Looks good to me!

**Comment by mxosman** [comment]
I second Jen's LGTM! Thank you so much, Ian!

File: packages/design-system/src/components/GlobalStyle/GlobalStyle.ts [hunk 1]
```diff
@@ -16,9 +16,11 @@
...
 import { rem } from "polished";
-import { palette, spacing } from "../../styles";
-import "@fontsource/public-sans";
+import { palette, spacing, typography } from "../../styles";
+import "@fontsource/public-sans/400.css";
 import "@fontsource/public-sans/500.css";
+import "@fontsource/public-sans/600.css";
+import "@fontsource/public-sans/700.css";
 import "@fontsource/libre-franklin";
 import "@fontsource/libre-franklin/500.css";
```
Comment (ohaibbq) [inline packages/design-system/src/components/GlobalStyle/GlobalStyle.ts:25]: Can these be dropped?

**Review by ohaibbq — APPROVED** [review]
So good!!

File: packages/design-system/src/components/GlobalStyle/GlobalStyle.ts (hunk 1 above)
Comment (macfarlandian) [reply packages/design-system/src/components/GlobalStyle/GlobalStyle.ts:25]: oh yes I believe so, that was an oversight. Good catch!

**Comment by macfarlandian** [comment]
this update has passed both design and code review in Spotlight and Pulse Dashboard so I am going to proceed with the merge!

**Pull request merged**